### PR TITLE
vendor: github.com/moby/buildkit v0.10.4

### PR DESCRIPTION
**Pull request by thaJeztah**
release notes: https://github.com/moby/buildkit/releases/tag/v0.10.4

full diff: https://github.com/moby/buildkit/compare/8e2d9b9006ca...v0.10.4

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
Lol updating BuildKit, and getting failiures on the classic builder. I think it's just flakiness with that server; we could consider pushing a pre-built image, but pulling Windows images is also slow, so not sure if that would make things better.

```
---> ba80c7979f32
Step 9/13 : ADD https://frippery.org/files/busybox/busybox-w32-${BUSYBOX_VERSION}.exe /bin/busybox.exe
ADD failed: Get "https://frippery.org/files/busybox/busybox-w32-FRP-3329-gcf0fa4d13.exe": dial tcp 93.93.131.127:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

let me kick it again

**Comment by thaJeztah** [comment]
Oh! Other failure in BuildKit integration tests; coudl be a flake, but let me post, just in case 

```
=== CONT  TestIntegration/TestNamedImageContextTimestamps/worker=dockerd/frontend=client
dockerfile_test.go:5551:
Error Trace:	dockerfile_test.go:5551
run.go:88
run.go:202
Error:      	Received unexpected error:
exporter "image" could not be found
github.com/moby/buildkit/util/stack.Enable
/src/util/stack/stack.go:77
github.com/moby/buildkit/util/grpcerrors.FromGRPC
/src/util/grpcerrors/grpcerrors.go:188
github.com/moby/buildkit/util/grpcerrors.UnaryClientInterceptor
/src/util/grpcerrors/intercept.go:41
google.golang.org/grpc.(*ClientConn).Invoke
/src/vendor/google.golang.org/grpc/call.go:35
github.com/moby/buildkit/api/services/control.(*controlClient).Solve
/src/api/services/control/control.pb.go:1443
github.com/moby/buildkit/client.(*Client).solve.func2
/src/client/solve.go:208
golang.org/x/sync/errgroup.(*Group).Go.func1
/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57
runtime.goexit
/usr/local/go/src/runtime/asm_amd64.s:1581
failed to solve
github.com/moby/buildkit/client.(*Client).solve.func2
/src/client/solve.go:221
golang.org/x/sync/errgroup.(*Group).Go.func1
/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57
runtime.goexit
/usr/local/go/src/runtime/asm_amd64.s:1581
Test:       	TestIntegration/TestNamedImageContextTimestamps/worker=dockerd/frontend=client
sandbox.go:235: stdout: /usr/bin/dockerd
sandbox.go:235: stderr: /usr/bin/dockerd
sandbox.go:238: > startCmd 2022-08-24 15:09:10.758741918 +0000 UTC m=+81.159107005 [/usr/bin/dockerd --data-root /tmp/integration1341243162/dybz2rofa7t2j/root --exec-root 
```

**Comment by thaJeztah** [comment]
hmm.. GitHub actions went AWOL? don't merge!

**Comment by thaJeztah** [comment]
@tonistiigi @jedevc looks like CI continues failing on one of the BuildKit tests; could one of you have a look if it's genuine, or some flaky test?

**Comment by thaJeztah** [comment]
Discussing with @tonistiigi - looks like there may be a test that's either broken (not compatible with current docker engine), or an actual issue; if it's the test, we need to disable it in buildkit (and override the version of the buildkit integration tests we run)

**Comment by thaJeztah** [comment]
Something similar as the override that was removed in https://github.com/moby/moby/commit/e05f614267213c93bd941d8a8980a0265a2e4634

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -128,7 +128,9 @@ jobs:
         name: BuildKit ref
         run: |
           ./hack/go-mod-prepare.sh
-          echo "BUILDKIT_REF=$(./hack/buildkit-ref)" >> $GITHUB_ENV
+# FIXME(thaJeztah) temporarily overriding version to use for tests; see https://github.com/moby/moby/pull/44028#issuecomment-1225964929
+#          echo "BUILDKIT_REF=$(./hack/buildkit-ref)" >> $GITHUB_ENV
+          echo "BUILDKIT_REF=8e2d9b9006caadb74c1745608889a37ba139acc1" >> $GITHUB_ENV
```
Comment (thaJeztah) [inline .github/workflows/ci.yml:133]: Added this temporary override

**Comment by thaJeztah** [comment]
I'll bring this one in; will create a ticket or PR in BuildKit to skip that test on dockerd

**Comment by crazy-max** [comment]
> I'll bring this one in; will create a ticket or PR in BuildKit to skip that test on dockerd

Might bring this commit https://github.com/moby/buildkit/pull/3003/commits/5d4e9382840e5f93118be12702f631fca8bcfd63

**Comment by thaJeztah** [comment]
@crazy-max that's on your fork? GitHub doesn't show a PR attached to it; yes, we should probably cherry-pick that one into the v0.10 branch

**Comment by crazy-max** [comment]
@thaJeztah This one https://github.com/moby/buildkit/pull/3003

**Comment by thaJeztah** [comment]
Ah, right; looks like that was stalling on the other bits in that PR; perhaps makes sense to open it as a separate one

**Pull request merged**